plugins/gateway/tyk/v3.2.2/plugin.go

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2021 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"unsafe"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/TykTechnologies/tyk/ctx"
+	"github.com/TykTechnologies/tyk/log"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/apiclarity/apiclarity/plugins/api/client/client"
+	"github.com/apiclarity/apiclarity/plugins/api/client/client/operations"
+	"github.com/apiclarity/apiclarity/plugins/api/client/models"
+)
+
+var logger = log.Get()
+
+// Called during post phase for setting the apiDefinition since we dont get it in the response phase.
+func PostGetAPIDefinition(rw http.ResponseWriter, r *http.Request) {
+	apiDefinition := ctx.GetDefinition(r)
+	if apiDefinition == nil {
+		apiDefinition = apiDefinitionRetriever(r.Context())
+	}
+
+	if apiDefinition == nil {
+		logger.Error("Failed to get api definition")
+		return
+	}
+	ctx.SetDefinition(r, apiDefinition)
+}
+
+// Called during response phase
+func ResponseSendTelemetry(rw http.ResponseWriter, res *http.Response, req *http.Request) {
+	logger.Info("Handling telemetry")
```

Comment:
is it easy to configure log level?

Reply:
Not sure. the info level is printed by default from what I saw (on my deployment)

Reply:
If so not sure we need to print detailed print on every transaction
maybe only in debug - but then need to understand how to toggle debug mode (:

Reply:
It is not a detailed print

Reply:
sorry my bad - thought that it printed the full telemetry